internal/build/build.go

```diff
@@ -403,11 +405,14 @@ func linkMainPkg(ctx *context, pkg *packages.Package, pkgs []*aPackage, llFiles
 		}
 	}()
 
-	// add rpath
+	// add rpath and find libs
 	exargs := make([]string, 0, ctx.nLibdir<<1)
+	libs := make([]string, 0, ctx.nLibdir*3)
```

Comment:
Don't need save libs. Can use `debug/macho` to find all dylibs (see [dylibdeps](https://github.com/goplus/llgo/blob/main/chore/dylibdeps/dylibdeps.go)) and exclude those in `/lib/` or `/usr/lib`.

Reply:
> exclude those in `/lib/` or `/usr/lib`.

My first thought was to apply rpath only to explicitly linked libs (those `-l` args). I'm not sure if using an exclude list is a good idea, considering not only `/lib` and `/usr/lib` need consideration, but also many other system-level libdirs like `/System/Library/Frameworks` and `/Library/Frameworks`, and possibly others.

Reply:
It's strange that we use prefix match to find real path of a dylib. If we want to do this, it seem using `os.Readlink` is better.

Reply:
```go
if dylib, ok := load.(*macho.Dylib); ok {
	if strings.HasPrefix(filepath.Base(dylib.Name), fmt.Sprintf("lib%s.", lib)) {
		return dylib.Name
	}
}
```

The `dylib.Name` here refers to the dylib ID.

Mach-O typically requires the dylib ID format to be `/path/to/lib<name>.<version>.dylib`. For our purposes, only the last path element needs to be matched, as it is unique (`clang` can ensures this for us). So this is why I do `filepath.Base(dylib.Name)`. And the prefix match is enough because the `<version>.dylib` part is not important is this case.

If we do fully match `os.Readlink(dylib.Name)`, then we may need to simulate `clang`'s process of searching `-L` to find `-l` to obtain the specific absolute path. I'm not sure if this is worth doing, since our use case here is pretty straight.